lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -340,9 +340,7 @@ public function delete_item( $request ) {
 			return new WP_Error( 'rest_user_cannot_delete_post', __( 'Sorry, you are not allowed to delete this post.' ), array( 'status' => 401 ) );
 		}
 
-		$get_request = new WP_REST_Request( 'GET', rest_url( 'wp/v2/' . $this->get_post_type_base( $this->post_type ) . '/' . $post->ID ) );
-		$get_request->set_param( 'context', 'edit' );
-		$response = $this->prepare_item_for_response( $post, $get_request );
+		$response = rest_do_request( 'GET', '/wp/v2/' . $this->get_post_type_base( $this->post_type ) . '/' . $post->ID, array( 'context' => 'edit' ) );
```

Comment:
It would be nice if I could pass any arbitrary path, including one prepared with `rest_url()`